src/Breadcrumbs.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Honeybadger;
+
+use Honeybadger\Support\EvictingQueue;
+
+class Breadcrumbs extends EvictingQueue
+{
+    /**
+     * @param $item
+     *
+     * @return self
+     */
+    public function add($item)
+    {
+        $item = [
+            'message' => (string) $item['message'],
+            'category' => (string) ($item['category'] ?? 'custom'),
+            'metadata' => $this->sanitize($item['metadata'] ?? []),
+            'timestamp' => (int) ($item['timestamp'] ?? time()),
+        ];
+
+        return parent::add($item);
+    }
+
+    /**
+     * Limit an array to a simple [key => value] (one level) containing only primitives.
+     */
+    private function sanitize(array $data): array
+    {
+        $sanitized = [];
+        foreach ($data as $key => $value) {
+            if (is_array($value) || is_object($value) || is_resource($value)) {
+                continue;
```

Comment:
Rather than remove the items entirely, how about we add them with the string value `"[DEPTH]"`, which is how we indicate that a max-depth was reached (see [general-purpose sanitizer in the Ruby gem](https://github.com/honeybadger-io/honeybadger-ruby/blob/master/lib/honeybadger/util/sanitizer.rb#L18)).

Reply:
Done.